pkg/kapp/diff/changes_set_with_periodic_rs.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"fmt"
+	"time"
+
+	ctlres "github.com/vmware-tanzu/carvel-kapp/pkg/kapp/resources"
+)
+
+type ChangeSetWithPeriodicRes struct {
+	existingRs, newRs []ctlres.Resource
+}
+
+func NewChangeSetWithPeriodicRs(existingRs, newRs []ctlres.Resource) *ChangeSetWithPeriodicRes {
+	return &ChangeSetWithPeriodicRes{existingRs: existingRs, newRs: newRs}
+}
+
+func (d ChangeSetWithPeriodicRes) Calculate() error {
+	existingRs := existingVersionedResources(d.existingRs)
+	existingVersionRsGrouped := newGroupedVersionedResources(existingRs.Versioned).groupedRes
+
+	return d.checkAndUpdateResWithMaxDurationAnn(existingVersionRsGrouped, existingRs)
+}
+
+func (d ChangeSetWithPeriodicRes) checkAndUpdateResWithMaxDurationAnn(existingVersionRsGrouped map[string][]ctlres.Resource,
```

Comment:
I think we can move this to `Calculate()`?

Reply:
Renamed  api to Prepare as suggested above and moved the changes to it.